python/docs/dlis/investigation.rst

```diff
@@ -0,0 +1,171 @@
+
+* Build dlisio_ and lfp_ in debug mode.
+* Start Python debugging process, find failing C++ call, start gdb (or
+  similar C++ debugger) and connect to Python process.
+* While debugging further keep in mind that dlisio uses layered-file-protocol
+  library. It means that all the offsets passed inside dlisio are not absolute
+  (TIF or rp66 header bytes are not included), so you'll have to dig into lfp_
+  library to find out what really happens.
+* You might need to modify the code to give yourself more anchor points if
+  failure, for example, happens late in the long loop.
+
+
+Figure out failure reason
+-------------------------
+
+Results of your investigation might be something that makes sense - certain
+specification violation, missing bytes or some specific but consistent between
+several files error.
+
+Sometimes, however, it won't be possible to figure out what exactly went wrong.
+Then failure can't be classified and there is little that can be fixed.
+
+
+Check VR boundaries
+^^^^^^^^^^^^^^^^^^^
+
+Search for the first bytes ``FF 01`` before and after offset. Check that
+record length matches distance between the tells (if file is tiffed, be
+aware of tif header bytes which will affect the distance).
+In most cases VR header will be ``20 00 FF 01``. If it is not, look
+closer. What you see might still be a valid record header. Or it can be random
+bytes which happened to be ``FF 01``.
+
+
+Check LRS
+^^^^^^^^^
+
+Find LRS to which failing tell belongs by manually counting logical record
+segments from the start of the VR. Pay attention to the header (attributes and
+code). If attributes or code seem to be random, something is likely spoiled
+already. EFLR expected codes are [``0x00 - 0x0B``], IFLR ones are ``0x00`` and
+``0x01``. Attribute byte pattern often is ``xxx00000``, but can differ.
+
+Make sure that all the attributes make sense and create consistent record.
+Check the padbytes length: it might be too big and "eat" part of the data.
+
+Check attributes in template/object
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For EFLRs find the object that fails and manually check object attributes
+based on object set template. You might find that some objects are missing data
+or that there are major specification violations.
+
+Check fdata
+^^^^^^^^^^^
+
+For IFLRs find failing frame. Make sure that you see frame name near the
+start of the record. It might be difficult to check the format
+string and all the data, but you can compare failing frame record length with
+previous frame record length. More often than not they should be the same. You
+might also check the frame numbers, they are usually sequential.
+
+Fixes
+-----
+In certain cases you might be able to fix the file.
+
+Remember that there is no guarantee that your fix will lead to the correct file:
+
+* fix itself might be wrong
+* there might already be too much data missing
+* fix might create screwed image of the file
+* there might be more errors further down the file
```

Comment:
Would remove this too. It's all very obvious? I don't see how this adds any value